lib/charms/mongodb_libs/v0/mongodb_provider.py

```diff
@@ -133,6 +158,142 @@ def oversee_users(self, departed_relation_id: Optional[int], event):
                 logger.info("Drop database: %s", database)
                 mongo.drop_database(database)
 
+    ################################################################################
+    # LEGACY RELATIONS
+    ################################################################################
+    def _on_legacy_relation_created(self, event):
+        """Legacy relations for MongoDB opporate without a password and so we update the server accordingly and
+        set a flag.
+        """
+        logger.warning("DEPRECATION WARNING - `mongodb` interface is a legacy interface.")
+
+        # TODO, future PR check if there are any new relations that are related to this charm and will be effected
+        # by loss of password. If so go into blocked state.
+
+        # TODO, future PR check if already running without auth
+        self._stop_mongod_service()
+        self._update_mongod_service(auth=False)
+        self._start_mongod_service()
```

Comment:
Should you handle the `BlockedStatus` here to prevent some undesired call?

Reply:
yes good catch 🙏